code/parachain/frame/assets-transactor-router/src/tests/routing.rs

```diff
@@ -0,0 +1,1092 @@
+#![allow(clippy::disallowed_methods)] // allow unwrap() in tests
+use crate::mocks::{AssetId, RuntimeOrigin};
+
+use composable_traits::{
+	assets::{AssetInfo, BiBoundedAssetName, BiBoundedAssetSymbol, CreateAsset},
+	xcm::assets::XcmAssetLocation,
+};
+use mocks::{new_test_ext, Test};
+
+use crate::*;
+
+const NATIVE_ASSET_ID: AssetId = 1;
+const ACCOUNT_NATIVE: u128 = 1;
+const ACCOUNT_LOCAL: u128 = 2;
+const ACCOUNT_FOREIGN: u128 = 3;
+const ACCOUNT_TO: u128 = 4;
+
+// creates for routing 1 local asset and 1 foreign asset(native asset is specified in config)
+fn create_assets() -> (AssetId, AssetId) {
+	let protocol_id_local = *b"testloca";
+	let nonce_local = 0;
+	let protocol_id_foreign = *b"testfore";
+	let nonce_foreign = 0;
+	let asset_info_local = AssetInfo {
+		name: Some(
+			BiBoundedAssetName::from_vec(b"local asset".to_vec()).expect("string is within bound"),
+		),
+		symbol: None,
+		decimals: Some(12),
+		ratio: None,
+		existential_deposit: 100,
+	};
+	let asset_id_local =
+		Pallet::<Test>::create_local_asset(protocol_id_local, nonce_local, asset_info_local)
+			.unwrap();
+
+	let foreign_asset_id = XcmAssetLocation(xcm::v2::MultiLocation::parent());
+	let foreign_asset_info = AssetInfo {
+		name: Some(
+			BiBoundedAssetName::from_vec(b"Kusama".to_vec()).expect("string is within bound"),
+		),
+		symbol: Some(
+			BiBoundedAssetSymbol::from_vec(b"KSM".to_vec()).expect("string is withing bound"),
+		),
+		decimals: Some(12),
+		ratio: None,
+		existential_deposit: 1000,
+	};
+
+	let asset_id_foreign = Pallet::<Test>::create_foreign_asset(
+		protocol_id_foreign,
+		nonce_foreign,
+		foreign_asset_info,
+		foreign_asset_id,
+	)
+	.unwrap();
+	(asset_id_local, asset_id_foreign)
+}
+
+// issue assets to different accounts and in different amount
+fn mint_assets(asset_id_local: AssetId, asset_id_foreign: AssetId) {
+	Pallet::<Test>::mint_into(RuntimeOrigin::root(), asset_id_local, ACCOUNT_LOCAL, 1000).unwrap();
+	Pallet::<Test>::mint_into(RuntimeOrigin::root(), asset_id_foreign, ACCOUNT_FOREIGN, 2000)
+		.unwrap();
+	Pallet::<Test>::mint_into(RuntimeOrigin::root(), NATIVE_ASSET_ID, ACCOUNT_NATIVE, 3000)
+		.unwrap();
```

Comment:
these numbers should be extracted into module level constants so you can reference them in the tests, rather having magic numbers in the assertions.